<launch>
    <arg name="control_topic" default="/car/mux/ackermann_cmd_mux/input/navigation" />
    <arg name="init_pose_topic" default="/initialpose" />
    <arg name="plan_file" default="$(find mushr_ros_intro)/plans/planned.txt" />
    <arg name="dist_thresh" default='2.0' />
    <arg name="angle_thresh" default='45' />
    <arg name="max_speed" default='2.0' />
    <arg name="min_speed" default='0.5' />
    <arg name="steer" default='1' />


    <arg name="map_server" default="1"/>
    <arg name="map_file" default="$(find mushr_mujoco_ros)/maps/real-floor0.yaml" />

    <node pkg="mushr_ros_intro" type="path_publisher.py" name="path_publisher" output="screen">
        <param name="control_topic" value="$(arg control_topic)" />
        <param name="init_pose_topic" value="$(arg init_pose_topic)" />
        <param name="plan_file" value="$(arg plan_file)" />
        <param name="dist_thresh" value="$(arg dist_thresh)" />
        <param name="angle_thresh" value="$(arg angle_thresh)" />
        <param name="max_speed" value="$(arg max_speed)" />
        <param name="min_speed" value="$(arg min_speed)" />
        <param name="steer" value="$(arg steer)" />
    </node>

    <!-- Launch  map server-->
    <group if="$(arg map_server)">
        <include file="$(find mushr_sim)/launch/map_server.launch">
            <arg name="map" value="$(arg map_file)" />
        </include>
    </group>

</launch>
